src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -245,6 +245,20 @@ class ConstraintsChecker
         return CheckConstraintMaxValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T, typename U, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current,
+                                 const chip::app::DataModel::Nullable<U> & expected)
```

Comment:
Two things:
1) We don't need he "not enum" thing here, because we're not making use of it in any way.
2) The code in the function will only compile if T == U, because otherwise you can't compare the nullables, so we might as well just make that clea.

```suggestion
    template <typename T>
    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current,
                                 const chip::app::DataModel::Nullable<T> & expected)
```

Reply:
Updated.